pxtcompiler/emitter/service.ts

```diff
@@ -1088,6 +1093,13 @@ namespace ts.pxtc.service {
                         tsPos = smallest.ts.startPos
                     }
                 }
+
+                // filter based on word match if we get too many (ideally we'd leave this filtering for monaco as it's
+                // better at fuzzy matching and fluidly changing but for performance reasons we want to do it here)
+                if (!isMemberCompletion && resultSymbols.length > MAX_SYMBOLS_BEFORE_FILTER) {
+                    resultSymbols = resultSymbols
+                        .filter(s => (isPython ? s.symbol.pyQName : s.symbol.qName).indexOf(partialWord) >= 0)
```

Comment:
question; my understanding was that the result list always had to filter symbols down / it typically doesn't add back in symbols (I think for example this was why we don't show completions for things starting with `_`, as it was non trivial to 'add back in' those symbols). If that's the case, it seems like this could lose fidelity of completions / not accurately gather completions later on, if we're semi-arbitrarily filtering early. Is my understanding of this incorrect / am I missing context here?

(as a sidenote for anyone who doesn't have experience with that, we filter out symbols that start with `-` as they're seen as 'internal data' the user won't typically want to see, like sprites `._x` field / etc)

Reply:
Yes, for the most part we filter symbols down over time, but we also do sometimes add new symbols (like for member completion, local vars, or keywords) back in, so it should be that it's only global APIs we're filtering this way.

We definitely lose some fidelity compared to before like "gm" won't show "game.*" members any more, but I think the trade off for performance is worth it.